### DISCUSSION TITLE
Lets say I used backtesting.py and my algo is Great how would I make this in the real world

### DISCUSSION THREAD

**Discussion by LegitCamper** [kernc/backtesting.py#827] [General]
A quote I see in this community a lot is "The best Algorithm implemented in the wrong way is worthless". My concern at the moment is even when I get an algorithm that is very profitable making a new program to trade that does match my backtest 100% will not perform the same. Is there a way to retrofit the backtesting class to use it in live trading?

**Comment by texasraj** [comment]
thats a good question.. in the same boat.. need to see how to translate SL and TP orders into real world orders with the broker..

**Comment by kernc** [comment]
> Is there a way to retrofit the backtesting class to use it in live trading?

Current progress is as far as https://github.com/kernc/backtesting.py/issues/81 (i.e. stale). So you'd have to rewrite the logic for trading and accept minor deviations in results, relative to the quality of your model.

With brokers, SL-TP order pair is represented as OCO orders, e.g.:
https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Order.is_contingent
https://www.google.com/search?q=oco+ccxt
https://interactivebrokers.github.io/tws-api/oca.html

**Reply by mattastrophe** [reply]
I'm also interested in extending backtest.py to support live trading.  Have you considered integrating with JackRabbitRelay and gaining access to lots of brokers in one shot?

https://github.com/rapmd73/JackrabbitRelay
